jerry-core/jerry.c

```diff
@@ -2121,6 +2162,85 @@ jerry_foreach_object_property (const jerry_value_t obj_val, /**< object value */
 } /* jerry_foreach_object_property */
 
 /**
+ * Common part for jerry_resolve_promise and jerry_reject_promise.
+ *
+ * @return undefined value - if success
+ *         value marked with error flag - otherwise
+ */
+static jerry_value_t
+jerry_resolve_or_reject_promise (jerry_value_t promise,
+                                 jerry_value_t argument,
+                                 bool is_resolve)
```

Comment:
> why just directly call the resolve/reject in the branch?

Code space savings. You'd have 2x function calls vs 1x.
Often you already have the `bool should_call_resolve` one way or another, for example in your snippet, there is `some_condition`. In this case, `some_condition == !should_call_resolve`.